### PR TITLE
k8s - add support for Server Side apply

### LINKED ISSUE #87: Server-side apply support
##### SUMMARY
It would be great if `k8s` module supported server-side apply.
https://kubernetes.io/docs/reference/using-api/server-side-apply/
https://kubernetes.io/blog/2020/04/01/kubernetes-1.18-feature-server-side-apply-beta-2/

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
`k8s` module

### PR DESCRIPTION
**Pull request by abikouo**
##### SUMMARY

[Server side apply](https://kubernetes.io/docs/reference/using-api/server-side-apply/) is now support for ``k8s`` module with this Pull request.
The feature is not yet released on kubernetes-client, once this is done, we can merge this pull request.
closes #87 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

k8s

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/k8s.py [hunk 1]
```diff
@@ -152,6 +152,26 @@
     - mutually exclusive with C(name).
     type: str
     version_added: 2.3.0
+  server_side_apply:
+    description:
+    - When this option is set, apply runs in the server instead of the client.
+    - Ignored if C(apply) is not set or is set to False.
+    - This option requires "kubernetes > 18.20.0".
```
Comment (gravesm) [inline plugins/modules/k8s.py:159]:
```suggestion
    - This option requires "kubernetes >= 19.15.0".
```

File: plugins/module_utils/common.py [hunk 2]
```diff
@@ -1043,6 +1045,24 @@ def _empty_resource_list():
                         params = {}
                         if self.check_mode:
                             params["dry_run"] = "All"
+                        if server_side_apply:
+                            params["server_side"] = True
+                            if LooseVersion(kubernetes.__version__) <= LooseVersion(
+                                "18.20.0"
+                            ):
+                                msg = "kubernetes > 18.20.0 is required to use server side apply."
```
Comment (gravesm) [inline plugins/module_utils/common.py:1053]:
```suggestion
                            if LooseVersion(kubernetes.__version__) < LooseVersion(
                                "19.15.0"
                            ):
                                msg = "kubernetes >= 19.15.0 is required to use server side apply."
```

File: molecule/default/tasks/taint.yml [hunk 3]
```diff
@@ -227,6 +227,10 @@
       that:
       - _result.changed
 
+  - name: Display all taints
+    debug:
+      msg: "{{ taint_patch_1 }} + {{ taint_patch_2 }}"
+
```
Comment (gravesm) [inline molecule/default/tasks/taint.yml:233]: Can you revert this change?

Comment (abikouo) [reply molecule/default/tasks/taint.yml:233]: I do this to debug the CI which was no stable
Will remove

**Comment by abikouo** [comment]
recheck
